be1-go/hub/organizer_test.go

```diff
@@ -0,0 +1,326 @@
+	require.NoError(t, err)
+	require.Equal(t, laoChannel.rollCall.state, Created)
+	require.Equal(t, laoChannel.rollCall.id, string(dataCreate.ID))
+
+	// Open
+	dataOpen, err := newCorrectOpenRollCallData(laoID, dataCreate.ID, message.OpenRollCallAction)
+	require.NoError(t, err)
+	err = laoChannel.processRollCallObject(organizerKeyPair.publicBuf, dataOpen)
+	require.NoError(t, err)
+	require.Equal(t, laoChannel.rollCall.state, Open)
+	require.Equal(t, laoChannel.rollCall.id, string(dataOpen.UpdateID))
+
+	// Generate public keys
+	var attendees []message.PublicKey
+
+	for i := 0; i < 10; i++ {
+		keypair, err := generateKeyPair()
+		require.NoError(t, err)
+		attendees = append(attendees, keypair.publicBuf)
+	}
+
+	// Close
+	dataClose1, err := newCorrectCloseRollCallData(laoID, dataOpen.UpdateID, attendees[:8])
+	require.NoError(t, err)
+	err = laoChannel.processRollCallObject(organizerKeyPair.publicBuf, dataClose1)
+	require.NoError(t, err)
+	require.Equal(t, laoChannel.rollCall.state, Closed)
+	require.Equal(t, laoChannel.rollCall.id, string(dataClose1.UpdateID))
+	for _, attendee := range attendees[:8] {
+		_, ok := laoChannel.attendees[string(attendee)]
+		require.True(t, ok)
+	}
+
+	// Reopen
+	dataReopen, err := newCorrectOpenRollCallData(laoID, dataClose1.UpdateID, message.ReopenRollCallAction)
+	require.NoError(t, err)
+	err = laoChannel.processRollCallObject(organizerKeyPair.publicBuf, dataReopen)
+	require.NoError(t, err)
+	require.Equal(t, laoChannel.rollCall.state, Open)
+	require.Equal(t, laoChannel.rollCall.id, string(dataReopen.UpdateID))
+
+	// Close
+	dataClose2, err := newCorrectCloseRollCallData(laoID, dataReopen.UpdateID, attendees)
+	require.NoError(t, err)
+	err = laoChannel.processRollCallObject(organizerKeyPair.publicBuf, dataClose2)
+	require.NoError(t, err)
+	require.Equal(t, laoChannel.rollCall.state, Closed)
+	require.Equal(t, laoChannel.rollCall.id, string(dataClose2.UpdateID))
+	for _, attendee := range attendees {
+		_, ok := laoChannel.attendees[string(attendee)]
+		require.True(t, ok)
+	}
+}
+
+func TestOrganizer_CreateRollCallWrongID(t *testing.T) {
+	_, laoChannel, err := createLao(oHub, organizerKeyPair)
+	require.NoError(t, err)
+
+	// create the roll call
+	id := []byte{1}
+	dataCreate := newCreateRollCallData(id, getTime(), "my roll call")
+	err = laoChannel.processRollCallObject(organizerKeyPair.publicBuf, dataCreate)
+	require.Error(t, err)
+	require.Equal(t, string(laoChannel.rollCall.state), "")
+	require.Equal(t, string(laoChannel.rollCall.id), "")
+}
+
+func TestOrganizer_CreateRollCallWrongSender(t *testing.T) {
+	laoID, laoChannel, err := createLao(oHub, organizerKeyPair)
+	require.NoError(t, err)
+
+	keypair, err := generateKeyPair()
+	require.NoError(t, err)
+
+	// Create the roll call
+	dataCreate, err := newCorrectCreateRollCallData(laoID)
+	require.NoError(t, err)
+	err = laoChannel.processRollCallObject(keypair.publicBuf, dataCreate)
```

Comment:
This is not working in the continuous integration. Perhaps you could use type composition to create a type similar to publicKey which implements the methods of the interface so that you can keep this as is.

Reply:
Thank you for the review! I updated the tests now I think it is working.